app/scripts/inpage.js

```diff
@@ -121,3 +121,69 @@ function restoreContextAfterImports () {
     console.warn('MetaMask - global.define could not be overwritten.')
   }
 }
+
+/* ------ HANDLE INJECTING PLUGINS ------- */
+
+var plugins = {};
+
+try {
+var rand = function(){ return Math.ceil(Math.random()*10000) }
+// the following should be dynamic, NOT hard coded,
+// based on a UI a user can use to add directories or URLs.
+// We DO NOT want to have to add dependencies to MetaMask for plugins,
+// but for now, I did, just to get the gulp build system working properly.
+plugins.GUN = require('gun/gun.min.js');
+plugins.SEA = require('gun/sea.js');
+
+// NOTE: SEA should actually be core, not considered a plugin!
+// It is what other plugins would use as their metamask crypto API.
+
+// move this to its own file...
+var n1 = ['PROOF OF WORK', 'ENCRYPTION', 'DECRYPTION', 'SIGNING', 'SIGNATURE VERIFICATION', 'SECRET SHARING'];
+['work', 'encrypt', 'decrypt', 'sign', 'verify', 'secret'].forEach(function(method, i){
+  plugins.SEA[method] = function(data, pair, cb, opt){
+    console.log("METAMASK HAS HIJACKED SEA's "+n1[i]+" FOR SECURITY REASONS!");
+    // NOTE: SEA's official API is callback, not Promise. 
+    return new Promise(function(res, rej){ // Should be callback style, temporary for now.
+      cb = cb || function(){};
+      pair = (pair instanceof Function)? undefined : pair;
+      var put = {data: data, pair: pair, opt: opt};
+      inpageProvider.sendAsync({jsonrpc:'2.0', method: 'SEA.'+method, put: put}, function(a,b,c){
+        if(b.err || b.error){
+          console.log(b.err || b.error);
+          plugins.SEA.err = b.err || b.error;
+          cb(); res(); // rej() is handled by SEA, not the plugin.
+          return;
+        }
+        cb(b.ack); res(b.ack);
+    })});
+  }
+});
+var n2 = ['NAMING', 'KEY GENERATION'];
+['name', 'pair'].forEach(function(method, i){
+  plugins.SEA[method] = function(cb, opt){
+    console.log("METAMASK HAS HIJACKED SEA's "+n2[i]+" FOR SECURITY REASONS!");
+    // NOTE: SEA's official API is callback, not Promise. 
+    return new Promise(function(res, rej){ // Should be callback style, temporary for now.
+      cb = cb || function(){};
+      var get = {opt: opt};
+      inpageProvider.sendAsync({jsonrpc:'2.0', method: 'SEA.'+method, get: get}, function(a,b,c){
```

Comment:
Nit: The current syntax for the "web3 provider rpc" is to namespace with an underscore. so `SEA_$METHOD`.

Reply:
k, if that helps makes a difference, I'll try to include that with the console change.